### ISSUE TITLE
PodDisruptionBudget calculation is not working

### ISSUE THREAD

**Issue by s3than** [hashicorp/consul-helm#265]
As per #71 the latest version of helm https://github.com/helm/helm/releases/tag/v2.15.0 includes the following `Updated Sprig to 2.20.0`

This causes the previous fix to pod disruption budget to return -1 now on a new installation.

Updating the _helpers.tpl to the following resolves however if you run it in helm < 2.15.0 it's broken again..

I'll hopefully have a fix shortly that works in both and will advise.

```
{{- define "consul.pdb.maxUnavailable" -}}
{{- if eq (int .Values.server.replicas) 1 -}}
{{ 0 }}
{{- else if .Values.server.disruptionBudget.maxUnavailable -}}
{{ .Values.server.disruptionBudget.maxUnavailable -}}
{{- else -}}
{{- if eq (int .Values.server.replicas) 3 -}}
{{- 1 -}}
{{- else -}}
{{- div (int .Values.server.replicas) 2 -}}
{{- end -}}
{{- end -}}
{{- end -}}
```

**Comment by s3than** [comment]
Resolution is same as #264

**Comment by lkysow** [comment]
Yeah looks like helm has reverted this change in their latest release (v2.15.1): https://github.com/helm/helm/releases/tag/v2.15.1

I don't think there's a good workaround on our side other than asking users to update Helm.

**Comment by s3than** [comment]
Updated pull request to just include tests, happy to close as resolved when using 2.15.1 of helm
